swagger.yaml

```diff
@@ -581,6 +581,87 @@ paths:
           required: true
           schema:
             type: string
+      responses:
+        201:
+          description: Success.
+        400:
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+        401:
+          description: Unauthorized.
+          schema:
+            $ref: '#/definitions/infra_error'
+        403:
+          description: Forbidden.
+          schema:
+            $ref: '#/definitions/infra_error'
+        404:
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+        405:
+          description: Method Not Allowed.
+          schema:
+            $ref: '#/definitions/error'
+        500:
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+
+  /clusters/{cluster_id}/ignition-config:
+    get:
+      tags:
+        - installer
+      summary: Get the cluster ignition config
+      operationId: GetClusterIgnitionConfig
+      parameters:
+        - in: path
+          name: cluster_id
+          type: string
+          format: uuid
+          required: true
+      responses:
+        200:
+          description: Success.
+          schema:
+            type: string
+        401:
+          description: Unauthorized.
+          schema:
+            $ref: '#/definitions/infra_error'
+        403:
+          description: Forbidden.
+          schema:
+            $ref: '#/definitions/infra_error'
+        404:
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+        405:
+          description: Method Not Allowed.
+          schema:
+            $ref: '#/definitions/error'
+        500:
+          description: Error.
+          schema:
+            $ref: '#/definitions/error'
+    patch:
+      tags:
+        - installer
+      summary: Override values in the initial ignition config
+      operationId: UpdateClusterIgnitionConfig
+      parameters:
+        - in: path
+          name: cluster_id
+          type: string
+          format: uuid
+          required: true
+        - in: body
+          name: ignition-config-params
+          required: true
+          schema:
+            type: string
```

Comment:
Better to have this as a proper object rather than string, it makes changing the API easier in the future if necessary.

Reply:
The json spec for ignition v3.1 is defined [here](https://github.com/coreos/ignition/blob/master/config/v3_1/schema/ignition.json) do we want to duplicate all of that here or just a subset?

Reply:
Sorry I meant an object with a single property, that string.  That way if we want to add more parameters in the future we can.

Reply:
Ah, perfect :+1: